### PR TITLE
add support for OCSP_SINGLERESP functions

### PR DESCRIPTION
**Pull request by samuel40791765**
### Issues:
Addresses `CryptoAlg-2420`

### Description of changes: 
Add support for ASN1 and X509v3 extension functions for `OCSP_SINGLERESP`. These are all consumed in Ruby's OCSP module.

### Call-outs:
N/A

### Testing:
New tests for the extension functions and `OCSP_SINGLERESP` creation.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: crypto/ocsp/ocsp_extension.c [hunk 1]
```diff
@@ -37,6 +37,19 @@ X509_EXTENSION *OCSP_BASICRESP_get_ext(OCSP_BASICRESP *bs, int loc) {
   return X509v3_get_ext(bs->tbsResponseData->responseExtensions, loc);
 }
 
+int OCSP_SINGLERESP_add_ext(OCSP_SINGLERESP *sresp, X509_EXTENSION *ex,
```
Comment (andrewhop) [inline crypto/ocsp/ocsp_extension.c:40]: Missing null checks.

Comment (samuel40791765) [reply crypto/ocsp/ocsp_extension.c:40]: Yeah I was on the fence for these, they're single calls against existing functions that already have null checks. 
https://github.com/aws/aws-lc/blob/34aceaf04b1ff0499104a73e67c7d4b36551f1f8/crypto/x509/x509_v3.c#L67-L124

Should we add them here? I'm ok either way.

Comment (WillChilds-Klein) [reply crypto/ocsp/ocsp_extension.c:40]: if there's no way for a NULL pointer passed in by the caller to segfault, then IMO we can skip redundant null checks.

**Review by WillChilds-Klein — COMMENTED** [review]
other than null checks, this PR looks good to me

**Review by WillChilds-Klein — APPROVED** [review]

**Review by andrewhop — APPROVED** [review]

**Pull request merged**